### ISSUE TITLE
Cross-slice re-export not detected

### ISSUE THREAD

**Issue by conarti** [conarti/eslint-plugin-feature-sliced#27]
## Problem

The plugin does not detect cross-slice re-exports, which violate FSD principles.

## Example

```typescript
// src/features/policy-management/model/types.ts
export { PolicyVerb, AffinityKind } from "@/entities/policies";
```

This re-exports symbols from `entities/policies` slice through `features/policy-management` slice. Consumers of `features/policy-management` would import these symbols from the wrong slice, breaking FSD layering.

The expected behavior is to import directly from the source slice:

```typescript
// consumer.ts
import { PolicyVerb } from "@/entities/policies"; // correct
import { PolicyVerb } from "@/features/policy-management"; // incorrect - cross-slice re-export
```

## Expected

The plugin should warn when a module re-exports symbols from another slice (cross-layer or same-layer), as this creates hidden coupling and breaks FSD import hierarchy.

## Current behavior

No warning is emitted. The `absolute-relative` rule catches incorrect path styles within a slice, but not cross-slice re-exports.

## Plugin version

`@conarti/eslint-plugin-feature-sliced@2.0.0-rc.6`

**Comment by conarti** [comment]
Decision for 2.1.0.

Today `export ... from` is not analysed at all, so a re-export can bypass the layer rules while the equivalent import is reported. This changes in 2.1.0:

- a re-export from a higher layer or from a sibling slice is reported exactly like the equivalent import;
- a re-export of a lower layer through a slice, the case in this issue, is also reported by default, and a rule option turns it off.

`export * from`, `export * as ns from` and type-only re-exports count as dependencies too, while `@x` public API paths stay allowed.

For the record: the FSD documentation states no rule about re-exports of other slices, and Steiger does not report them either. The second item is therefore deliberately stricter than the current upstream tooling, which is why it comes with an option.
